core/src/main/java/org/apache/spark/shuffle/sort/ShuffleInMemorySorter.java

```diff
@@ -69,8 +72,15 @@ public int numRecords() {
     return pos;
   }
 
-  public void reset() {
+  public long shrinkMemory() {
```

Comment:
This function is the only thing we need to change, can we still call it `reset` (other do not need to change the caller)?